Fix black cell backgrounds in scheduler

willgearty opened the pull request:
This fixes the coloring of cells (and text color) in the AJAX Scheduler for programs with class IDs with fewer than 4 digits. The new logic pads the class IDs with zeros to get at least 2 digits to work with, then uses the last 2 digits of the resulting IDs. ~~Therefore, programs with class IDs with fewer than 3 digits can expect very blue/green colors, but red is incorporated once class IDs have at least 3 digits.~~

Before:
![image](https://user-images.githubusercontent.com/7232514/56085166-9cc36d00-5df3-11e9-964b-df3bf9b83401.png)

After:
![image](https://user-images.githubusercontent.com/7232514/56461044-2679ba00-6361-11e9-8d9e-bd608c35672e.png)

Fixes #2361.

kkbrum commented:
This looks fine, but maybe instead of padding with 0s and getting lots of blues we could just do if statements based on the number of digits in the class code and for the ones with less digits, calculate colors in some smart way such that the colors are more fun? I don't really like that the colors are going to be so sad until programs get 100 classes :(

willgearty commented:
OK, I've changed the approach to use a pre-established palette of 100 colors and to rotate through them based on the last two digits of the class code (using a prime number and modulus to make consecutive classes have pretty different colors). Let me know what you think! (I've updated the image in the OP)
